website/static/js/addonNodeConfig.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Module that controls the addon node settings. Includes Knockout view-model
+ * for syncing data, and HGrid-folderpicker for selecting a folder.
+ */
+'use strict';
+
+var ko = require('knockout');
+require('knockout.punches');
+var $ = require('jquery');
+var Raven = require('raven-js');
+
+var ZeroClipboard = require('zeroclipboard');
+ZeroClipboard.config('/static/vendor/bower_components/zeroclipboard/dist/ZeroClipboard.swf');
+var $osf = require('js/osfHelpers');
+var oop = require('js/oop');
+var FolderPickerViewModel = require('js/folderPickerNodeConfig');
+
+ko.punches.enableAll();
+
+var AddonFolderPickerViewModel = oop.extend(FolderPickerViewModel, {
+    constructor: function(addonName, url, selector, folderPicker, opts) {
+        var self = this;
+        self.super.constructor.call(self, addonName, url, selector, folderPicker);
+        // whether the auth token is valid
+        self.validCredentials = ko.observable(true);
+        // Emails of contributors, can only be populated by activating the share dialog
+        self.emails = ko.observableArray([]);
+        self.loading = ko.observable(false);
+        // Whether the contributor emails have been loaded from the server
+        self.loadedEmails = ko.observable(false);
+        // externalAccounts
+        self.accounts = ko.observable([]);
+        
+        // List of contributor emails as a comma-separated values
+        self.emailList = ko.pureComputed(function() {
+            return self.emails().join([', ']);
+        });
+        self.selectedFolderType = ko.pureComputed(function() {
+            var userHasAuth = self.userHasAuth();
+            var selected = self.selected();
+            return (userHasAuth && selected) ? selected.type : '';
+        });
+        self.messages.SUBMIT_SETTINGS_SUCCESS =  ko.pureComputed(function() {
+            return 'Successfully linked "' + $osf.htmlEscape(self.folder().name) + '". Go to the <a href="' +
+                self.urls().files + '">Files page</a> to view your content.';
+        });
+        
+        // Overrides
+        self.options = {
+            onPickFolder: function(evt, item) {
+                evt.preventDefault();
+                var name = item.data.path !== '/' ? item.data.path : '/ (Full ' + self.addonName + ')';
+                self.selected({
+                    name: name,
+                    path: item.data.path,
+                    id: item.data.id
+                });
+                return false; // Prevent event propagation
+            }
+        };
+        // Overrides
+        self.options = $.extend(self.options, opts);
```

Comment:
Why assign self.options then override it with a mutation of itself?

How about:

```js
var defaults = {
    onPickFolder: ....
}
self.options = $.extend({}, defaults, opts);
```